### ISSUE TITLE
[Shaman] Flurry refreshing bug

### ISSUE THREAD

**Issue by zapsuu** [SunwellTracker/issues#2864]
Decription: Enchancement talent Flurry refresh charges wrong. item ID: http://db.darkwizard.pl/?spell=16283

How it works: If you score a critical hit from melee (auto attacks and abilites like Stormstrike/Lava Lash/Windfury) you receive 3 charges, and if you score a critical hit with the buff active you get only 2 charges. With active buff critcal hit just consume 1 stack. It's easy to check using 2H weapon.

How it should work: Flurry should grant 3 charges any time a critical strike is scored.

Source (you should point out proofs of your report, please give us some source):

**Comment by Nikrotip** [comment]
The bug is still active! Whenever 'Flurry' is reapplied with Flurry stacks already present it adds 2 instead of 3 fresh stacks.

**Comment by Nikrotip** [comment]
Proof link: https://www.youtube.com/watch?v=fvkN9-C1X3k&feature=youtu.be

**Comment by Nikrotip** [comment]
The bug still occurs

**Comment by Hankmoodyx** [comment]
Okay, I want to provide more proof and also in my opinion useful information, that the flurry is not working correctly and also ask is there any progress in fixing it, because it really punishes enhancement scaling with items, beacuse additional crit % is not giving any additional Flurry uptime, when this talent is bugged.
So Like was told in the previous posts, If Flurry is up either on 1 or 2 stacks, melee crits are not refreshing it to 3 stacks, but to 2 stacks. I tested it right now and Special attack crits, like Stormstrike, Lava Lash and Windfury Weapon proc refreshes it correctly to 3 stacks, but melee crits doesn't. SO there is an issue only with melee attacks in that case.
I wanna post few screenshots with data from Retail Trial of the Crusader and Flurry uptime at close to similar gear (atleast in terms of critical strike chance, which is the most important thing in Flurry uptime Obviously). 
02.11.2009 Twin Valkyr's kill and enhancement data: http://www.worldoflogs.com/reports/rt-mn3oe7m7jmc3so3t/details/2/?s=253&e=564
![obraz](https://user-images.githubusercontent.com/64852186/86829822-2b757700-c095-11ea-8ed0-c77fe46f4b38.png)
Sadly u can't check his exact critical chance on attacks, which can proc Flurry, because of the absorb dmg is showing in the damage breakdown.
Anub'arak kill data from the same player and same report:
![obraz](https://user-images.githubusercontent.com/64852186/86830388-ec93f100-c095-11ea-9940-c9242ca4b7eb.png)
Now we see insane 96% Flurry uptime. His average crit on attacks which can proc Flurry (which are Melee swings, Windfury procs, Lava Lash and Stormstrike to clarify) is at about 52%. 
Another example of different fight, more where u have 100% uptime at dmg without any intermissions, Lord Jarraxus fight: http://www.worldoflogs.com/reports/lt2k8upkqt2kqemi/analyze/dd/source/?s=2144&e=2413
![obraz](https://user-images.githubusercontent.com/64852186/86830842-76dc5500-c096-11ea-9a32-49b2b6bc30cf.png)
This shaman average crit on attacks I mentioned earlier was at about 55%.
Now lets see what are the numbers on Frosthold server:
Jarraxus 25hc: 82% flurry uptime, with about 55% crit on proccing attacks.
![obraz](https://user-images.githubusercontent.com/64852186/86831705-85773c00-c097-11ea-9cd3-1a3691a02f4c.png)
Jarraxus 25nm (it's more burn fight with lower aoe spells prio, so in theory I was using Flurry proccing abilities, which are working correctly, more often):
86% flurry uptime, with about 59% crit on proccing attacks.
![obraz](https://user-images.githubusercontent.com/64852186/86832233-3251b900-c098-11ea-9f2e-dfb656b02446.png)
Twin Valkyrs 25hc (I cut the last 15-20s of a fight because I died, so to be sure that this is the uptime from where I was alive 100% of the time): 84,7% flurry uptime 
![obraz](https://user-images.githubusercontent.com/64852186/86833845-5f06d000-c09a-11ea-9a82-000f3d6a95cd.png)
So in conclusion those numbers are kinda far from which were on retail. 
Also the flurry mechanic is also in Fury Warriors talents and I will check it later, if it's working correctly or not and update this post.
I hope that this problem will be fixed as soon as possible, because it's still bugged after 5 months from posting.

**Comment by DiomacheTharghan** [comment]
Up, any news?

**Comment by Selot1234** [comment]
@Afgann ping

**Comment by eleclai** [comment]
Bug is still Up. Having a DPS loss for free feels so good.

When you already have stacks and proc Flurry, you'll be put at 2 Total Stacks, resulting in a lower uptime of the buff, as you have 1 fewer "chance" of refreshing it without losing the effect. 

An average 10% of Uptime lost on this buff is A big damage loss for the class.

I hope @Afgann Fixes this soon, as he Self-Assigned 5 months ago.

One Enha Shaman can only Hope...

**Comment by DiomacheTharghan** [comment]
![1DE5BC7D-1C3D-4165-9BA2-E43AB10676A9_cx0_cy3_cw0_w1023_r1_s](https://user-images.githubusercontent.com/44828921/89474830-fc7f1d80-d786-11ea-9b51-d8e626738f63.jpg)

**Comment by DiomacheTharghan** [comment]
![lol2](https://user-images.githubusercontent.com/44828921/89474860-099c0c80-d787-11ea-9481-23fc2c79cdfb.jpg)

**Comment by DiomacheTharghan** [comment]
![lol2](https://user-images.githubusercontent.com/44828921/89474907-29cbcb80-d787-11ea-98b1-8f1582cde3f4.jpg)

**Comment by Tiddlywank** [comment]
Just commenting to confirm that the bug is still up, and the bug affects both Enhancement Shamans and Fury Warriors. If you have the Flurry buff, critting only gains you 2 stacks, tested on a dummy pretty easily.

**Comment by Hankmoodyx** [comment]
Okay for everyone who was interested in that, bug seems to be fixed, I didn't see any changelog mention about that on Sunwell discord, I noticed that after todays raid, apparently flurry seems to work as it should, tested that on dummy, and also average raid uptime went up for 10%+ which is really nice.

**Comment by DiomacheTharghan** [comment]
@Hankmoodyx confirmed
